notification2-examples/java-examples/src/main/java/c8y/example/notification/samples/PlatformConfiguration.java

```diff
@@ -0,0 +1,72 @@
+package c8y.example.notification.samples;
+
+import c8y.example.notification.common.platform.SubscriptionRepository;
+import c8y.example.notification.common.platform.TokenService;
+import com.cumulocity.model.authentication.CumulocityCredentialsFactory;
+import com.cumulocity.sdk.client.Platform;
+import com.cumulocity.sdk.client.PlatformBuilder;
+import lombok.Getter;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.FileBasedConfiguration;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.configuration2.builder.FileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.fluent.Parameters;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+
+@Getter
+public class PlatformConfiguration {
+
+    private static PlatformConfiguration platformConfiguration;
+
+    private static final String CUMULOCITY_BASE_URL = "c8y.baseURL";
+    private static final String CUMULOCITY_TENANT = "c8y.tenant";
+    private static final String CUMULOCITY_USERNAME = "c8y.username";
+    private static final String CUMULOCITY_PASSWORD = "c8y.password";
+
+    private static final String CUMULOCITY_SOURCE_ID = "c8y.source.id";
+    private static final String CUMULOCITY_NOTIFICATION2_WS_URL = "c8y.notification2.websocket.url";
+    private static final String CUMULOCITY_NOTIFICATION2_SUBSCRIBER = "c8y.notification2.subscriber";
+
+    private final String websocketUrl;
+
+    private final Platform platform;
+
+    private final TokenService tokenService;
+    private final SubscriptionRepository subscriptionRepository;
+
+    private PlatformConfiguration() throws ConfigurationException {
+        final Configuration configuration = new FileBasedConfigurationBuilder<FileBasedConfiguration>(
+                PropertiesConfiguration.class)
+                .configure(new Parameters()
+                        .properties()
+                        .setFileName("notifications-example.properties"))
+                .getConfiguration();
```

Comment:
Isn't that overcomplicating? I mean this should be just a basic example so I would just create simple class with all fields hardcoded directly in it.

Reply:
My idea was to provide platform parameters via a file and example specific one via Class directly as I don't want one to be dealing with the former often.